typegen/src/utils.rs

```diff
@@ -419,13 +835,38 @@ mod tests {
                 }
             };
         }
+        macro_rules! nested_typeB {
+            ($ty:ident, $generic:ty, $inner:ty) => {
+                struct $ty;
+                impl scale_info::TypeInfo for $ty {
+                    type Identity = Self;
+                    fn type_info() -> scale_info::Type {
+                        scale_info::Type {
+                            path: Path::new("NestedType", "my::module"),
+                            type_params: vec![TypeParameter::new(
+                                "B",
+                                Some(meta_type::<$generic>()),
+                            )],
+                            type_def: TypeDef::Composite(TypeDefComposite::new([Field::new(
+                                None,
+                                meta_type::<$inner>(),
+                                None,
+                                Vec::new(),
+                            )])),
+                            docs: vec![],
+                        }
+                    }
+                }
+            };
+        }
 
         // A and B are the same because generics explain the param difference.
         //
         //NestedType<T = u32>(u32)
         //NestedType<T = bool>(bool)
         nested_type!(A, u32, u32);
         nested_type!(B, bool, bool);
+        nested_typeB!(G, u64, u64);
```

Comment:
This `nested_typeB` just renames the generic type param `B` instead of `T` which nested_type! is using?

Reply:
done